### ISSUE TITLE
The escape sequence interferes with evil-lisp-state

### ISSUE THREAD

**Issue by razcampagne** [syl20bnr/spacemacs#578]
Hi,

I changed the evil-escape sequence keys to "ht" on my configuration and found out the command bound to "h" didn't work anymore in evil-lisp-state. I looked in the code and it seems the shadowed function is not properly set when setting evil-escape for evil-lisp-state. The following patch fix the bug (tell me if you prefer a pull request or if I overlooked something)

```
diff --git a/spacemacs/extensions/evil-escape/evil-escape.el b/spacemacs/extensions/evil-escape/evil-escape.el
index 404a238..1d313ef 100644
--- a/spacemacs/extensions/evil-escape/evil-escape.el
+++ b/spacemacs/extensions/evil-escape/evil-escape.el
@@ -203,7 +203,12 @@ with a key sequence."
                                     :delete-func isearch-delete-char))
   ;; lisp state if installed
   (eval-after-load 'evil-lisp-state
-    '(eval '(evil-escape-define-escape "lisp-state" evil-lisp-state-map evil-normal-state)))
+    '(progn
+       (setq evil-escape-lisp-state-shadowed-func
+             (lookup-key evil-lisp-state-map (evil-escape--first-key)))
+       (eval `(evil-escape-define-escape "lisp-state" evil-lisp-state-map
+                                         evil-normal-state
+                                         :shadowed-func ,evil-escape-lisp-state-shadowed-func))))
   ;; iedit state if installed
   (eval-after-load 'evil-iedit-state
     '(progn
```

**Comment by syl20bnr** [comment]
Nice, PR welcome ! Also if you have time you can submit to https://github.com/syl20bnr/evil-escape as well. (for now I keep it as an extension in spacemacs hence the duplication).

**Comment by syl20bnr** [comment]
Fixed by #585
